cpp/include/experimental/graph_view.hpp

```diff
@@ -140,27 +140,29 @@ class partition_t {
     auto major_first =
       hypergraph_partitioned_
         ? vertex_partition_offsets_[row_comm_size_ * partition_idx + row_comm_rank_]
-        : vertex_partition_offsets_[row_comm_rank_ * col_comm_size_];
+        : vertex_partition_offsets_[col_comm_rank_ * row_comm_size_];
     auto major_last =
       hypergraph_partitioned_
         ? vertex_partition_offsets_[row_comm_size_ * partition_idx + row_comm_rank_ + 1]
-        : vertex_partition_offsets_[(row_comm_rank_ + 1) * col_comm_size_];
+        : vertex_partition_offsets_[(col_comm_rank_ * row_comm_size_)+1];
 
+    std::cout<<"CCR: "<<col_comm_rank_<<" RCS: "<<row_comm_size_<<std::endl;
+    std::cout<<"INDEX: "<<((row_comm_rank_ * col_comm_size_)+1)<<std::endl;
     return std::make_tuple(major_first, major_last);
   }
 
   vertex_t get_matrix_partition_major_first(size_t partition_idx) const
   {
     return hypergraph_partitioned_
              ? vertex_partition_offsets_[row_comm_size_ * partition_idx + row_comm_rank_]
-             : vertex_partition_offsets_[row_comm_rank_ * col_comm_size_];
+             : vertex_partition_offsets_[col_comm_rank_ * row_comm_size_];
   }
 
   vertex_t get_matrix_partition_major_last(size_t partition_idx) const
   {
     return hypergraph_partitioned_
              ? vertex_partition_offsets_[row_comm_size_ * partition_idx + row_comm_rank_ + 1]
-             : vertex_partition_offsets_[(row_comm_rank_ + 1) * col_comm_size_];
+             : vertex_partition_offsets_[(col_comm_rank_ * row_comm_size_)+1];
```

Comment:
This should be (col_comm_rank_ + 1) * row_comm_size

Reply:
and for the minor_range, minor_first, minor_last

```
  std::tuple<vertex_t, vertex_t> get_matrix_partition_minor_range() const
  {
    auto minor_first = get_matrix_partition_minor_first();
    auto minor_last  = get_matrix_partition_minor_last();

    return std::make_tuple(minor_first, minor_last);
  }

  vertex_t get_matrix_partition_minor_first() const
  {
    return hypergraph_partitioned_ ? vertex_partition_offsets_[col_comm_rank_ * row_comm_size_]
                                   : vertex_partition_offsets_[row_comm_rank_ * col_comm_size_];
  }

  vertex_t get_matrix_partition_minor_last() const
  {
    return hypergraph_partitioned_
             ? vertex_partition_offsets_[(col_comm_rank_ + 1) * row_comm_size_]
             : vertex_partition_offsets_[(row_comm_rank_ + 1) * col_comm_size_];
  }
```

Reply:
done